tests/unit/test_utils.py

```diff
@@ -1107,3 +1109,101 @@ def test_version_compare(x, y, result):
 )
 def test_version_compare_client(x, y, result):
     assert nxdrive.utils.version_compare_client(x, y) == result
+
+
+@pytest.mark.parametrize(
+    "default_config, conf_name, env, config_dump",
+    [
+        (
+            {},
+            "config_test.ini",
+            "features",
+            {
+                "feature_direct_edit": True,
+                "feature_auto_update": False,
+                "feature_s3": False,
+                "feature_direct_transfer": False,
+            },
+        ),
+        (
+            {"DEFAULT": {"env": "DEBUG"}},
+            "config_test.ini",
+            "features",
+            {
+                "feature_direct_edit": True,
+                "feature_auto_update": False,
+                "feature_s3": False,
+                "feature_direct_transfer": False,
+            },
+        ),
+        (
+            {
+                "DEFAULT": {"env": "DEBUG"},
+                "features": {
+                    "feature_direct_edit": False,
+                    "feature_auto_update": False,
+                    "feature_s3": False,
+                    "feature_direct_transfer": False,
+                    "feature_plop": True,
+                },
+            },
+            "config_test.ini",
+            "features",
+            {
+                "feature_direct_edit": True,
+                "feature_auto_update": True,
+                "feature_s3": True,
+                "feature_direct_transfer": True,
+            },
+        ),
+        (
+            {
+                "features": {
+                    "feature_direct_edit": False,
+                    "feature_auto_update": True,
+                    "feature_s3": True,
+                    "feature_direct_transfer": True,
+                    "feature_plop": True,
+                }
+            },
+            "config_test.ini",
+            "features",
+            {
+                "feature_direct_edit": True,
+                "feature_auto_update": False,
+                "feature_s3": False,
+                "feature_direct_transfer": False,
+            },
+        ),
+    ],
+)
+def test_save_config(default_config, conf_name, env, config_dump):
+    if default_config:
+        config = configparser.ConfigParser()
+        for key, value in default_config.items():
+            config[key] = value
+        with open(Options.nxdrive_home / conf_name, "w") as test_file:
+            config.write(test_file)
+
+    conf_path = nxdrive.utils.save_config(conf_name, env, config_dump)
+
+    assert conf_path == Options.nxdrive_home / conf_name
+    assert conf_path.is_file()
+
+    config = configparser.ConfigParser()
+    config.read(conf_path)
+
+    if default_config:
```

Comment:
Could you add comments to explain what does each code block ? It is too dense as-is.